File: 卡码/代码随想录/JS/贪心/01-简单/LeetCode_0860.js
//860. 柠檬水找零
const lemonadeChange = function(bills) {
	let five=0,ten=0;
	for(let bill of bills){
		switch(bill){
			case 5:
				five++;
				break;
			case 10:
				if(five>0){
					five--;
					ten++;
				}else return false;
				break;
			case 20:
				if(ten>0&&five>0){
					ten--;
					five--;
				}else if(five>2){
					five-=3;
				}else return false;
		}
	}
	return true;
};

File: 卡码/代码随想录/1/08-贪心算法/Go/01-简单/LeetCode_0860.go
//860. 柠檬水找零
package main

func lemonadeChange(bills []int) bool {
	five,ten:=0,0
	for _,bill:=range bills {
		switch bill {
			case 5:
				five++
			case 10:
				if five>0 {
					five--
					ten++
				}else{
					return false
				}
			case 20:
				if ten>0&&five>0 {
					ten--
					five--
				}else if five>2 {
					five-=3
				}else{
					return false
				}
		}
	}
	return true
}
